src/python/pants/backend/python/goals/export.py

```diff
@@ -78,9 +77,69 @@ def debug_hint(self) -> str | None:
         return self.resolve_name
 
 
+@rule_helper
+async def _do_export(
+    requirements_pex: Pex,
+    pex_pex: PexPEX,
+    dest: str,
+    resolve_name: str,
+    qualify_path_with_python_version: bool,
+) -> ExportResult:
+    # Get the path to the interpreter, and the full python version (including patch #).
+    res = await Get(
+        ProcessResult,
+        PexProcess(
+            pex=requirements_pex,
+            description="Get interpreter path and version",
+            argv=[
+                "-c",
+                "import sys; print(sys.executable); print('.'.join(str(x) for x in sys.version_info[0:3]))",
+            ],
+            extra_env={"PEX_INTERPRETER": "1"},
+        ),
+    )
+    interpreter_path, py_version = res.stdout.strip().decode().split("\n")
+
+    # NOTE: We add a unique prefix to the pex_pex path to avoid conflicts when multiple
```

Comment:
Should probably materialize the Pex PEX once, to some location under .pants.d say, rather than once, in a temporary location, per export? But that can wait.